terraform/modules/kubernetes/monitoring.tf

```diff
@@ -115,3 +115,48 @@ resource "azurerm_monitor_metric_alert" "redis_memory" {
     ]
   }
 }
+data "azurerm_monitor_diagnostic_categories" "default" {
+  count = var.deploy_azure_backing_services ? 1 : 0
+
+  resource_id = azurerm_postgresql_flexible_server.postgres-server[0].id
+}
+
+resource "azurerm_log_analytics_workspace" "pg_loganalytics" {
+  count = var.deploy_azure_backing_services ? 1 : 0
+  name                = "${azurerm_postgresql_flexible_server.postgres-server[0].name}-pgsql-logs"
+  location            = data.azurerm_resource_group.app-resource-group.location
+  resource_group_name = data.azurerm_resource_group.app-resource-group.name
+  sku                 = "PerGB2018"
+  retention_in_days   = 30
+}
+resource "azurerm_monitor_diagnostic_setting" "pg_loganalytics_diagnostic" {
+  count = var.deploy_azure_backing_services ? 1 : 0
+  name                       = "${azurerm_postgresql_flexible_server.postgres-server[0].name}-pgsql-diag"
+  target_resource_id         = azurerm_postgresql_flexible_server.postgres-server[0].id
+  log_analytics_workspace_id = azurerm_log_analytics_workspace.pg_loganalytics[0].id
+  storage_account_id         = azurerm_storage_account.pgsql_la_storage[0].id
+  dynamic "log" {
+    for_each = data.azurerm_monitor_diagnostic_categories.default[0].log_category_types
+    content {
+      category = log.value
+      enabled  = true
+
+      retention_policy {
+        enabled = true
+        days    = 30
```

Comment:
As discussed, there is no point setting the retention it in diagnostic settings since we only use log analytics and this setting won't apply

Reply:
removed retention period